h/static/scripts/group-forms/components/EditGroupMembersForm.tsx

```diff
@@ -4,25 +4,58 @@ import {
   Scroll,
   TrashIcon,
   IconButton,
+  Select,
 } from '@hypothesis/frontend-shared';
 import { useContext, useEffect, useMemo, useState } from 'preact/hooks';
 
 import { Config } from '../config';
 import type { APIConfig, Group } from '../config';
-import ErrorNotice from './ErrorNotice';
-import FormContainer from './forms/FormContainer';
-import type { GroupMembersResponse } from '../utils/api';
+import type { GroupMember, GroupMembersResponse, Role } from '../utils/api';
 import { callAPI } from '../utils/api';
+import FormContainer from './forms/FormContainer';
 import { pluralize } from '../utils/pluralize';
+import ErrorNotice from './ErrorNotice';
 import GroupFormHeader from './GroupFormHeader';
 import WarningDialog from './WarningDialog';
 
 type MemberRow = {
   username: string;
   userid: string;
   canDelete: boolean;
+  role: Role;
+  availableRoles: Role[];
 };
 
+/**
+ * Mappings between roles and labels. The keys are sorted in descending order
+ * of permissions.
+ */
+const roleStrings: Record<Role, string> = {
+  owner: 'Owner',
+  admin: 'Admin',
+  moderator: 'Moderator',
+  member: 'Member',
+};
+const possibleRoles: Role[] = Object.keys(roleStrings) as Role[];
+
+function memberToRow(member: GroupMember, currentUserid: string): MemberRow {
+  const role = member.roles[0] ?? 'member';
```

Comment:
Can you add a comment explaining why we pick only the first item in the `member.roles` list and ignore the rest?

IIRC, this was designed as a list to be more future-proof, allowing to add multiple roles without breaking the API contract, but for now all members will have only one role in that list.

Reply:
The UI design will need to change if we support more roles, and it isn't clear exactly how we'd present and enable selection of multiple roles. To keep things simple, we just assume a single role for the moment and will update this UI as necessary if we support multiple roles later.